ssl/s3_enc.c

```diff
@@ -537,7 +537,7 @@ int ssl3_digest_cached_records(SSL *s, int keep)
         }
 
         /* Loop through bits of algorithm2 field and create MD_CTX-es */
-        for (i = 0; ssl_get_handshake_digest(i, &mask, &md); i++) {
+        for (i = 0; ssl_get_certverify_digest(i, &mask, &md); i++) {
```

Comment:
The PRF MACs and the handshake MACs are specified independently. So for GOST2012-GOST8912-GOST8912, you have defined a PRF MAC of TLS1_PRF_GOST12_256, whilst you have additionally specified a handshake MAC of SSL_HANDSHAKE_MAC_GOST12_256. Similarly for GOST2012-NULL-GOST12.